astropy/coordinates/representation.py

```diff
@@ -216,17 +216,13 @@ def _values(self):
 
         The record array fields will have the component names.
         """
-        coordinates = [getattr(self, c) for c in self.components]
-        if NUMPY_LT_1_8:
-            # numpy 1.7 has problems concatenating broadcasted arrays.
-            coordinates = [(coo.copy() if 0 in coo.strides else coo)
-                           for coo in coordinates]
-
-        sh = self.shape + (1,)
-        dtype = np.dtype([(str(c), coo.dtype)
-                          for c, coo in zip(self.components, coordinates)])
-        return np.concatenate([coo.reshape(sh).value for coo in coordinates],
-                              axis=-1).view(dtype).squeeze()
+        # For PY2, convert possible unicode literal to string
+        coo_items = [(str(c) if six.PY2 else c, getattr(self, c))
```

Comment:
It used to do `str(c)` unconditionally which is guaranteed to work, and really the only reason I put this here was to be sure we would clean it up correctly for astropy 3.0. But on second thought, probably anybody doing that would do a grep for `PY2`, so then this would be found anyway.
Let me change it back...